### ISSUE TITLE
Dashboard accessibility

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#7301]
### Description
Accessibility notes for the CP dashboard at `/admin/dashboard`.

- The tab order gets a bit wacky when the widgets move around/get stacked at narrower screen sizes.
- The settings cog at the corner of each widget is only available on hover. Should be visible and keyboard accessible. If having the gear always visible is undesirable, perhaps have a single button that switches the dashboard into an editable mode where the gear becomes visible.
- There should be some indicator for links that open in a new tab (e.g. the blog posts in the "Craft News" widget).
- Craft support widgets:
  - Not keyboard operable **Related: #7290**
  - Closed state: icons have insufficient contrast and no text alternative. Could argue that the icons are decorative and don't need to be high contrast. Just need to make sure it's hidden to AT in that case.
  - Open state: a more obvious way to return to the original view, e.g. by adding a back/close button. I think the only way to close it right now is to use the escape key, which seems a bit unreliable when I press it after I've returned to the page from another tab.
  - Open state: the form fields should have persistent visible labels.
  - Open state: the form fields should have a description to indicate what is going to happen with the submission. Currently it's not clear until you get to the submit button that the form is meant to submits to public forums (StackOverflow or GitHub).
 
### Steps to reproduce
1. Go to `/admin/dashboard`

### Additional info
- Craft version: 3.5.17.1

**Comment by gcamacho079** [comment]
These issues should be mostly resolved in version 4.3. At this point, the tab order is consistent with the ordering of Dashboard widgets as set in the widget settings dropdown; this has been noted as a separate issue related to the masonry layout.
